upgrade/upgrade-scripts/upgrade

```diff
@@ -329,16 +331,30 @@ done
 shift $((OPTIND - 1))
 
 case "$1" in
-in-place)
+deferred)
+	UPGRADE_TYPE="${1^^}"
```

Comment:
nit: Today I learned a new obscure bash string substitution ;). I think it might be more readable to just put `UPGRADE_TYPE ="DEFERRED"` here though.

Reply:
Hehehe. I will change this.